download-utils/src/lib.rs

```diff
@@ -65,17 +66,9 @@ pub fn download_file<'a, 'b>(
 
     fs::create_dir_all(destination_file.parent().expect("parent"))
         .map_err(|err| err.to_string())?;
-
-    let mut temp_destination_file = destination_file.to_path_buf();
-    temp_destination_file.set_file_name(format!(
-        "tmp-{}",
-        destination_file
-            .file_name()
-            .expect("file_name")
-            .to_str()
-            .expect("to_str")
-    ));
-
+    let temp_destination_file = NamedTempFile::new()
```

Comment:
Can you append a random suffix to `temp_destination_file` instead.  We have snapshot cleanup code in particular that removes aborted tmp snapshots (`TMP_SNAPSHOT_PREFIX`), which will break with this change.

What's the actual issue you ran into BTW?  Seems weird to be trying to download the same file multiple times

Reply:
Multiple tests that run cargo build-bpf concurrently when they need to download a new version of bpf-tools.  I thought about using file locks, but it doesn't seem very straightforward to work with in Rust.

Reply:
I'll close this PR. I need to think how to avoid simultaneous downloads of the same file.

Reply:
Maybe use the `#[serial]` attribute for those tests?

Reply:
Yes, this is probably the best choice. Thank you for the suggestion. I was disappointed Rust test harness wouldn't allow to specify an order of execution of individual test cases or define dependencies between them, but didn't search 3rd party crates for this functionality.